### PR TITLE
MEV Tx Fees upgrade

### PR DESCRIPTION
**Pull request by arwer13**

### CODE REVIEW

File: contracts/0.4.24/Lido.sol [hunk 1]
```diff
@@ -78,6 +81,8 @@ contract Lido is ILido, IsContract, StETH, AragonApp {
     bytes32 internal constant BEACON_BALANCE_POSITION = keccak256("lido.Lido.beaconBalance");
     /// @dev number of Lido's validators available in the Beacon state
     bytes32 internal constant BEACON_VALIDATORS_POSITION = keccak256("lido.Lido.beaconValidators");
+    /// @dev total amount of Ether MEV and transaction rewards received by Lido contract
+    bytes32 internal constant MEV_TX_FEE_ETHER_POSITION = keccak256("lido.Lido.mevTxFeeEther");
```
Comment (arwer13) [inline contracts/0.4.24/Lido.sol:85]: add total 
write that these rewards are not accumulated in contract and not used for logic, it's just a counter

File: contracts/0.4.24/Lido.sol [hunk 2]
```diff
@@ -119,6 +124,16 @@ contract Lido is ILido, IsContract, StETH, AragonApp {
         _submit(0);
     }
 
+    /**
+    * @notice A payable function supposed to be funded only by LidoMevTxFeeVault contract
+    * @dev We need a separate function because funds received by default payable function
+    * will go through entire deposit algorithm
+    */
+    function mevTxFeeReceiver() external payable {
```
Comment (arwer13) [inline contracts/0.4.24/Lido.sol:132]: Renames
- Rename receiveMevTxFeeVault

File: contracts/0.4.24/interfaces/ILidoMevTxFeeVault.sol [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+pragma solidity 0.4.24;
+
+
+interface ILidoMevTxFeeVault {
+
+    /**
+    * @notice Withdraw all accumulated rewards to Lido contract
+    * @return amount uint256 of funds received as MEV and transaction fees in wei
+    */
+    function withdrawRewards() external returns (uint256 amount);
```
Comment (arwer13) [inline contracts/0.4.24/interfaces/ILidoMevTxFeeVault.sol:14]: comment that only Lido can withdraw

File: contracts/0.8.9/LidoMevTxFeeVault.sol [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+        LIDO = _lidoAddress;
+    }
+
+    /**
+    * @notice Withdraw all accumulated rewards to Lido contract
+    * @return amount uint256 of funds received as MEV and transaction fees in wei
+    */
+    function withdrawRewards() external returns (uint256 amount) {
+        require(msg.sender == LIDO, "Nobody except Lido contract can withdraw");
```
Comment (arwer13) [inline contracts/0.8.9/LidoMevTxFeeVault.sol:42]: Change require message format

**Comment by TheDZhon** [comment]
The code is adopter for another PR: #401

**Pull request closed without merging**